Lottie Android app crash when you open "Bullseye" because bullseye.json not found 404

Xiaomi Mi 9, Android 10. Log:

```
java.lang.IllegalStateException: Unable to parse composition
	at com.airbnb.lottie.LottieAnimationView$1.onResult(LottieAnimationView.java:74)
	at com.airbnb.lottie.LottieAnimationView$1.onResult(LottieAnimationView.java:67)
	at com.airbnb.lottie.LottieAnimationView$3.onResult(LottieAnimationView.java:91)
	at com.airbnb.lottie.LottieAnimationView$3.onResult(LottieAnimationView.java:84)
	at com.airbnb.lottie.LottieTask.notifyFailureListeners(LottieTask.java:158)
	at com.airbnb.lottie.LottieTask.access$200(LottieTask.java:27)
	at com.airbnb.lottie.LottieTask$1.run(LottieTask.java:133)
	at android.os.Handler.handleCallback(Handler.java:883)
	at android.os.Handler.dispatchMessage(Handler.java:100)
	at android.os.Looper.loop(Looper.java:224)
	at android.app.ActivityThread.main(ActivityThread.java:7551)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:539)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:950)
Caused by: java.lang.IllegalArgumentException: Unable to fetch https://raw.githubusercontent.com/airbnb/lottie-android/master/LottieSample/src/main/res/raw/bullseye.json. Failed with 404
404: Not Found

	at com.airbnb.lottie.network.NetworkFetcher.fetchFromNetworkInternal(NetworkFetcher.java:105)
	at com.airbnb.lottie.network.NetworkFetcher.fetchFromNetwork(NetworkFetcher.java:86)
	at com.airbnb.lottie.network.NetworkFetcher.fetchSync(NetworkFetcher.java:52)
	at com.airbnb.lottie.network.NetworkFetcher.fetchSync(NetworkFetcher.java:31)
	at com.airbnb.lottie.LottieCompositionFactory$1.call(LottieCompositionFactory.java:91)
	at com.airbnb.lottie.LottieCompositionFactory$1.call(LottieCompositionFactory.java:88)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:919)

```

--- a/lottie/src/main/java/com/airbnb/lottie/LottieImageAsset.java
+++ b/lottie/src/main/java/com/airbnb/lottie/LottieImageAsset.java
@@ -58,4 +58,11 @@ public class LottieImageAsset {
   public void setBitmap(@Nullable Bitmap bitmap) {
     this.bitmap = bitmap;
   }
+
+  /**
+   * Returns whether this asset has an embedded Bitmap or whether the fileName is a base64 encoded bitmap.
+   */
+  public boolean hasBitmap() {
+    return bitmap != null || (fileName.startsWith("data:") && fileName.indexOf("base64,") > 0);
+  }
 }

--- a/sample/src/main/kotlin/com/airbnb/lottie/samples/PlayerFragment.kt
+++ b/sample/src/main/kotlin/com/airbnb/lottie/samples/PlayerFragment.kt
@@ -429,6 +429,11 @@ class PlayerFragment : BaseMvRxFragment(R.layout.player_fragment) {
     private fun onCompositionLoaded(composition: LottieComposition?) {
         composition ?: return
 
+        // If the composition is missing any images, return the original image or null.
+        if (composition.images.any { (_, asset) -> !asset.hasBitmap() }) {
+            binding.animationView.setImageAssetDelegate { it.bitmap }
+        }
+
         binding.animationView.setComposition(composition)
         binding.controlBar.hardwareAccelerationToggle.isActivated = binding.animationView.layerType == View.LAYER_TYPE_HARDWARE
         binding.animationView.setPerformanceTrackingEnabled(true)

--- a/sample/src/main/kotlin/com/airbnb/lottie/samples/model/CompositionArgs.kt
+++ b/sample/src/main/kotlin/com/airbnb/lottie/samples/model/CompositionArgs.kt
@@ -3,6 +3,7 @@ package com.airbnb.lottie.samples.model
 import android.annotation.SuppressLint
 import android.net.Uri
 import android.os.Parcelable
+import kotlinx.android.parcel.IgnoredOnParcel
 import kotlinx.android.parcel.Parcelize
 
 @SuppressLint("ParcelCreator")
@@ -14,4 +15,7 @@ data class CompositionArgs(
         val asset: String? = null,
         val animationData: AnimationData? = null,
         val animationDataV2: AnimationDataV2? = null
-) : Parcelable
+) : Parcelable {
+        @IgnoredOnParcel
+        val isJson = (url ?: animationData?.lottieLink ?: animationDataV2?.file)?.endsWith("json") == true
+}

--- a/sample/src/main/res/layout/bullseye_activity.xml
+++ b/sample/src/main/res/layout/bullseye_activity.xml
@@ -9,7 +9,7 @@
         android:id="@+id/animationView"
         android:layout_width="256dp"
         android:layout_height="256dp"
-        app:lottie_url="https://raw.githubusercontent.com/airbnb/lottie-android/master/LottieSample/src/main/res/raw/bullseye.json"
+        app:lottie_rawRes="@raw/bullseye"
         app:lottie_autoPlay="true"
         app:lottie_loop="true"
         android:layout_gravity="center"/>

